### ISSUE TITLE
ESM builds

### ISSUE THREAD

**Issue by RReverser** [AssemblyScript/binaryen.js#37]
Would it be possible to provide ESM builds in addition to UMD ones? ES modules are pretty widely supported both in tooling as well as natively in browsers, and it would be nice to be able to use `binaryen` as an ES module in browser without polluting the global namespace or using Require.js or a bundler.

**Comment by dcodeIO** [comment]
I agree, that's something I'd like to have as well to switch AS to ESM eventually. Ideally a CMake task in the parent library. May even (switch to) emit ESM only, and we'd use a tool on this repo to transform it to UMD as a fallback.
